<launch>  
    <arg name="robot_name" default="locobot"/>
    <arg name="rviz_frame" default="$(arg robot_name)/odom  "/>
    <arg name="rvizconfig" default="$(find yolo_vision)/rviz/xslocobot_description_yolo.rviz" />
    
    <!-- Add YOLO node parameters -->
    <arg name="model_path" default="$(find yolo_vision)/models/best.pt"/>
    <arg name="image_topic" default="/$(arg robot_name)/camera/color/image_raw"/>
    
    <!-- Launch YOLO node -->
    <node name="yolo_node" pkg="yolo_vision" type="yolo_node.py" output="screen" ns="$(arg robot_name)">
        <param name="model_path" value="$(arg model_path)"/>
        <param name="image_topic" value="$(arg image_topic)"/>
    </node>

    <!-- Distance Detection Node -->
    <node name="distance_node" pkg="yolo_vision" type="distance_node.py" output="screen" ns="$(arg robot_name)">
        <!-- Topic configurations -->
        <param name="yolo_image_topic" value="camera/yolo/annotated_image"/>
        <param name="depth_image_topic" value="camera/depth/image_rect_raw"/>
        <param name="bbox_depth_topic" value="camera/yolo/bbox_depth"/>
        
        <!-- Node configurations -->
        <param name="queue_size" value="10"/>
        <param name="sync_slop" value="0.1"/>
        
        <!-- Color detection parameters -->
        <rosparam param="green_lower">[0, 255, 0]</rosparam>
        <rosparam param="green_upper">[0, 255, 0]</rosparam>
    </node>

      <!-- Original RViz node -->
    <node
        name="rviz"
        pkg="rviz"
        type="rviz"
        args="-f $(arg rviz_frame) -d $(arg rvizconfig)"
        ns="$(arg robot_name)">
        <remap from="/clicked_point" to="clicked_point"/>
        <remap from="/initialpose" to="initialpose"/>
        <remap from="/move_base_simple/goal" to="move_base_simple/goal"/>
    </node>

</launch>